### PR TITLE
[Merged by Bors] - feat(AlgebraicGeometry/Limits): Spec preserves finite coproducts

### PR DESCRIPTION
**Pull request by erdOne**
---

- [x] depends on: #14429

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/OpenImmersion.lean [hunk 1]
```diff
@@ -186,6 +186,24 @@ instance basic_open_isOpenImmersion {R : CommRingCat.{u}} (f : R) :
     exact Spec_map_localization_isIso R (Submonoid.powers f) x
 #align algebraic_geometry.Scheme.basic_open_IsOpenImmersion AlgebraicGeometry.Scheme.basic_open_isOpenImmersion
 
+instance {R} [CommRing R] (f : R) :
+    IsOpenImmersion (Spec.map (CommRingCat.ofHom (algebraMap R (Localization.Away f)))) :=
+  basic_open_isOpenImmersion (R := .of R) f
+
+lemma IsOpenImmersion.of_isLocalization {R S} [CommRing R] [CommRing S]
+    [Algebra R S] (f : R) [IsLocalization.Away f S] :
+    IsOpenImmersion (Spec.map (CommRingCat.ofHom (algebraMap R S))) := by
```
Comment (alreadydone) [inline Mathlib/AlgebraicGeometry/OpenImmersion.lean:195]: Haha, I was about to ask whether we have this :) A challenge (which I don't need) is to prove the same for localization at a f.g. submonoid.

Comment (erdOne) [reply Mathlib/AlgebraicGeometry/OpenImmersion.lean:195]: I don't think it is that hard actually. The same proof shows that the stalks are isomorphic and the underlying map is a topological embedding. To show that the image is open, we just need to show that it is the intersection of the basic open of the generators, which can't be that hard in lean right...?

Comment (alreadydone) [reply Mathlib/AlgebraicGeometry/OpenImmersion.lean:195]: Yeah probably not too hard. I was thinking about using induction and transitivity of open immersion and of localization.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by erdOne** [comment]
good to go now.

**Comment by mattrobball** [comment]
!bench

**Comment by erdOne** [comment]
!bench

**Comment by erdOne** [comment]
!bench

**Comment by erdOne** [comment]
I think this is the best I can do. @mattrobball any thoughts?

**Comment by mattrobball** [comment]
- Did you check there are no particular pain points (eg a tactic or instance synthesis that is mainly responsible)?
- Did the file get 40% longer?

**Comment by erdOne** [comment]
1. There are two defs that aren't instant but the slowness seems to be spread evenly through tactics.
2. The file length is almost 2x
So I think it's fine?

**Comment by mattrobball** [comment]
Yeah

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**